### PR TITLE
SONAR-6992 make sonar.core.id persistent and always a UUID

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  private static boolean isDate(String value) {
+    try {
+      new SimpleDateFormat("yyyyMMddHHmmss").parse(value);
+      return true;
+    } catch (ParseException e) {
+      return false;
+    }
+  }
+
+  private void ensureServerIdIsSet(@Nullable PropertyDto dto) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java]: MAJOR Make "ensureServerIdIsSet" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    checkState(dto != null, "Property %s is missing in database", SERVER_ID);
+    checkState(!isBlank(dto.getValue()), "Property %s is set but empty in database", SERVER_ID);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java]: MAJOR NullPointerException might be thrown as 'dto' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      new SimpleDateFormat("yyyyMMddHHmmss").parse(value);
+      return true;
+    } catch (ParseException e) {
+      return false;
+    }
+  }
+
+  private static void ensureServerIdIsSet(@Nullable PropertyDto dto) {
+    checkState(dto != null, "Property %s is missing in database", SERVER_ID);
+    checkState(!isBlank(dto.getValue()), "Property %s is set but empty in database", SERVER_ID);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java]: MAJOR NullPointerException might be thrown as 'dto' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.property.PropertyDto;
+import org.sonar.server.platform.cluster.Cluster;
+
+import static com.google.common.base.Preconditions.checkState;
+import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.sonar.api.CoreProperties.SERVER_ID;
+
+public class ServerIdManager implements Startable {
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java]: Yeaah, Master Seb created a class named "manager" ! :-)

Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdManager.java]: I'm definitely opened to suggestions on this one.
I lacked imagination last week, maybe something (anything!) better will come up.

**Comment by simonbrandhof** [comment]
LGTM

**Pull request merged**